dashboard/app/models/plc/user_course_enrollment.rb

```diff
@@ -74,6 +75,25 @@ def create_enrollment_unit_assignments
     end
   end
 
+  def create_authorized_teacher_user_permission
+    unless user.authorized_teacher?
+      UserPermission.create(
```

Comment:
Note that ActiveRecord `create` fails silently (the return indicates success). You may or may not want `create!`, which will throw an exception on failure.